src/core/filters/filterregistry.js

```diff
@@ -58,38 +58,43 @@ export default class FilterRegistry {
 
   /**
    * Gets the static filters as a {@link SimpleFilter|CombinedFilter} to send to the answers-core
+   *
    * @returns {CombinedFilter|SimpleFilter|null} Returns null if no filters with
    *                                             filtering logic are present.
    */
   getStaticFilterPayload () {
-    const filterNodes = this.getStaticFilterNodes();
-    return this._transformFilterNodes(filterNodes, FilterCombinator.AND);
+    const filterNodes = this.getStaticFilterNodes()
+      .filter(filterNode => {
+        return filterNode.getChildren().length > 0 || filterNode.getFilter().getFilterKey();
+      });
+    return filterNodes.length > 0
+      ? this._transformFilterNodes(filterNodes, FilterCombinators.AND)
+      : null;
   }
 
   /**
-   * Transforms a filter node {@link CombinedFilterNode} or {@link SimpleFilterNode} to
+   * Transforms a list of filter nodes {@link CombinedFilterNode} or {@link SimpleFilterNode} to
    * answers-core's {@link SimpleFilter} or {@link CombinedFilter}
    *
    * @param {Array<CombinedFilterNode|SimpleFilterNode>} filterNodes
    * @param {FilterCombinator} combinator from answers-core
-   * @returns {CombinedFilter|SimpleFilter|null} Returns null if no filters with
-   *                                             filtering logic are present.
+   * @returns {CombinedFilter|SimpleFilter} from answers-core
    */
   _transformFilterNodes (filterNodes, combinator) {
     const filters = filterNodes.flatMap(filterNode => {
       if (filterNode.children) {
         return this._transformFilterNodes(filterNode.children, filterNode.combinator);
       }
 
-      return !this._isEmpty(filterNode) && this._transformSimpleFilterNode(filterNode);
-    }).filter(filter => filter);
+      return this._transformSimpleFilterNode(filterNode);
+    });
 
-    return filters.length > 1
-      ? {
+    return filters.length === 1
+      ? filters[0]
+      : {
```

Comment:
If the length was zero, would we want to return what's below?

Reply:
I'm not sure! Technically, something like `{"$or":[]}` is valid to the API (it's the same as no filters), and we have a check in the caller, but I could move the check from `getStaticFilterPayload()`:
```
filterNodes.length > 0
      ? this._transformFilterNodes(filterNodes, FilterCombinators.AND)
      : null;
```
into this function to be safer!

Reply:
Okay! If the API knows how to interpret this correctly when it's `[]`, then I'm fine either way!